packages/ui/lib/api/api.js

```diff
@@ -0,0 +1,272 @@
+import FormType from "../enums/FormType";
+
+export default class API {
+  constructor(baseUrl) {
+    this.baseURL = baseUrl;
+
+    this.jwt = null;
+
+    this.endpointLogin = "/user/login";
+    this.endpointCreateUser = "/user/create";
+
+    this.endpointAuthorize = "/api/authorize";
+    this.endpointIntegration = (id) => `/api/integrations/${id}`;
+    this.endpointIntegrationConfigOptions = (id) =>
+      `${this.endpointIntegration(id)}/config/options`;
+    this.endpointIntegrations = "/api/integrations";
+    this.endpointIntegration = (id) => `/api/integrations/${id}`;
+    this.endpointSampleData = (id) => `/api/demo/sample/${id}`;
+    this.endpointIntegrationUserActionOptions = (id, action) =>
+      `/api/integrations/${id}/actions/${action}/options`;
+    this.endpointIntegrationUserActionSubmit = (id, action) =>
+      `/api/integrations/${id}/actions/${action}`;
+  }
+
+  setJwt(jwt) {
+    this.jwt = jwt;
+  }
+
+  async login(username, password) {
+    const params = {
+      username,
+      password,
+    };
+    return this._post(this.endpointLogin, params);
+  }
+
+  async createUser(username, password) {
+    const params = {
+      username,
+      password,
+    };
+    return this._post(this.endpointCreateUser, params);
+  }
+
+  // injects the access token into an object and returns the headers for most api calls
+  getHeaders() {
+    const headers = {
+      "Content-Type": "application/json",
+    };
+
+    if (this.jwt) {
+      headers.authorization = `Bearer ${this.jwt}`;
+    }
+
+    return headers;
+  }
+
+  // check the response of a fetch() before returning the data in JSON form.
+  // may throw an exception if the response.status corresponds to an error
+  async _checkResponse(response, url) {
+    if (response.status >= 400) {
+      console.error(
+        `Error: http [${response.status}] ${url}: ${JSON.stringify(response)}`
+      );
+    }
+
+    try {
+      if (response.headers.get("x-lh-set"))
```

Comment:
Reminds me (at some point) we should modify this to be `x-frigg-` etc... Task to create